src/main/java/io/ebeaninternal/server/core/DatabasePlatformFactory.java

```diff
@@ -115,24 +115,13 @@ private DatabasePlatform byDatabaseName(String dbName) {
    */
   private DatabasePlatform byDataSource(DataSource dataSource) {
 
-    Connection connection = null;
-    try {
-      connection = dataSource.getConnection();
+    try (Connection connection = dataSource.getConnection()) {
       DatabaseMetaData metaData = connection.getMetaData();
 
       return byDatabaseMeta(metaData, connection);
 
     } catch (SQLException ex) {
       throw new PersistenceException(ex);
-
-    } finally {
-      try {
-        if (connection != null) {
-          connection.close();
-        }
-      } catch (SQLException ex) {
-        logger.error(null, ex);
-      }
```

Comment:
@rbygrave I simplified the try catch blocks.
OLD: if connection.close throws an error, the error was logged
NEW: try-with-resources will catch that and a PersistenceException is thrown now.
Do you think this is OK?